### ISSUE TITLE
shift count is negative in int_to_fp_impl for arm64-apple-ios11.0-simulator

### ISSUE THREAD

**Issue by itf** [llvm/llvm-project#69779]
After this commit: https://github.com/llvm/llvm-project/commit/6dfea561ba96974b205c31546c5e2069429c75b1, 

I see the error:
```
/int_to_fp_impl.inc:60:29: error: shift count is negative [-Werror,-Wshift-count-negative]
   60 |   const int dstExpBias = (1 << (dstExpBits - 1)) - 1;
      |                             ^  ~~~~~~~~~~~~~~~~
1 error generated.
```
to reproduce:

```
clang -nostdlibinc compiler-rt/lib/builtins/floatuntitf.c -fsyntax-only -target arm64-apple-ios11.0-simulator
```

the error seems to be caused by wrongly assuming the size of a `long double`

**Comment by asb** [comment]
Thanks for the report! I think we need to move up `#if defined(CRT_HAS_TF_MODE)` before the int_to_fp_impl.inc include and defines. That target it seems doesn't support `__float128` and `long double` isn't a 128-bit float, so I think those helpers are never compiled.

It looks like int_to_fp.h also shouldn't be making the assumption that `long double` is a 128-bit float and have an ifdef similar to the one in fp_trunc.h. But I think that's separate to this bug.

**Comment by asb** [comment]
I believe this was fixed by 05a4212cc76d170445f5a118ec6d1a2a76f61fc8 (the original reported bug no longer reproduces). Please reopen if you still have issues.
